distributed/http/scheduler/tests/test_scheduler_http.py

```diff
@@ -93,23 +93,37 @@ async def test_prometheus(c, s, a, b):
 
     http_client = AsyncHTTPClient()
 
-    # request data twice since there once was a case where metrics got registered multiple times resulting in
-    # prometheus_client errors
-    for _ in range(2):
-        response = await http_client.fetch(
-            "http://localhost:%d/metrics" % s.http_server.port
-        )
+    async def fetch_metrics():
+        port = s.http_server.port
+        response = await http_client.fetch(f"http://localhost:{port}/metrics")
         assert response.code == 200
         assert response.headers["Content-Type"] == "text/plain; version=0.0.4"
 
         txt = response.body.decode("utf8")
         families = {
-            family.name: family for family in text_string_to_metric_families(txt)
+            family.name: family
+            for family in text_string_to_metric_families(txt)
+            if family.name.startswith("dask_scheduler_")
```

Comment:
This implies to me that worker metrics are now getting mixed in when you hit the scheduler port in tests? Probably because they're all running on the same async server?

I wonder if we need to use an async-friendly prometheus client, like https://github.com/hynek/prometheus-async or, more likely, https://github.com/claws/aioprometheus (since it seems to support creating multiple `Service` instances per process)?

Reply:
This is less about worker metrics (though I am not 100% sure that's not happening), but more about additional Python-related metrics that we collect but don't want to test for in here.

Reply:
> I wonder if we need to use an async-friendly prometheus client, like https://github.com/hynek/prometheus-async or, more likely, https://github.com/claws/aioprometheus (since it seems to support creating multiple Service instances per process)?

I'm open to it, but that's outside of the scope of this PR.

Reply:
> This implies to me that worker metrics are now getting mixed in when you hit the scheduler port in tests? Probably because they're all running on the same async server?

They shouldn't, we have different collectors on different server endpoints. They also use worker/scheduler-specific methods to gather metrics to display.

Reply:
Async clients look like an interesting thing but the actual value they provide is a bit unclear to me. 

I guess this can make the collection a bit smoother (currently it's a plain generator) such that it doesn't block the event loop during collection. Once we have more experience with this we can revisit this. Last time I was using prometheus I didn't encounter any loop instabilities